app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/demographics/ConvertAndSendLabOrderUsecase.java

```diff
@@ -21,7 +24,9 @@ public static ConvertAndSendLabOrderUsecase getInstance() {
 
     private ConvertAndSendLabOrderUsecase() {}
 
-    public void convertAndSend(Demographics<?> demographics) throws UnableToSendLabOrderException {
+    public void convertAndSend(Demographics<?> demographics)
+            throws UnableToSendLabOrderException, SecretRetrievalException, InvalidKeySpecException,
+                    NoSuchAlgorithmException {
```

Comment:
Maybe leave only the "UnableToSendLabOrderException" at the method level, then do a try/catch inside the method and throw an UnableToSendLabOrderException whenever any of the other three exceptions get caught, thoughts?

Reply:
I'm not familiar with Java best practices but this sounds right to me at a conceptual level. It makes sense to capture a high level UnableToSendLabOrderException in the convertAndSend method. Unless there's any debugging data that would get lost from the other exceptions. In that case, we could make sure to log the information before throwing UnableToSendLabOrderException